internal/journal/voljournal.go

```diff
@@ -649,6 +655,24 @@ func (conn *Connection) GetImageAttributes(ctx context.Context, pool, objectUUID
 	return imageAttributes, nil
 }
 
+// ReserveImageID reserves the image ID in omap
+func (conn *Connection) ReserveImageID(ctx context.Context, pool, reservedUUID, imageID string, cr *util.Credentials) error {
+	err := util.SetOMapKeyValue(ctx, conn.monitors, cr, pool, conn.config.namespace, conn.config.cephUUIDDirectoryPrefix+reservedUUID, conn.config.csiImageIDKey, imageID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetReservedImageID retrives the reserved image ID from the omap
+func (conn *Connection) GetReservedImageID(ctx context.Context, pool, reservedUUID string, cr *util.Credentials) (string, error) {
```

Comment:
Ideally we should be able to get this from `GetImageAttributes` (maybe once we have the fetch all keys in a single call).

Reply:
Tracked in the TODO list.